docroot/themes/humsci/humsci_basic/src/js/media/video-with-caption.js

```diff
@@ -0,0 +1,16 @@
+// In order to style the figcaption, figure elements have a display: table.
+// This causes issues when there is a video in a figure because the video no longer
+// fills the entire space of the container.
+// This JS sets a width of 100% to figures that contain videos.
+const videos = document.querySelectorAll('.field-media-oembed-video');
+
+if (videos) {
+  for (let i = 0; i < videos.length; i++) {
+    const video = videos[i];
+    const figure = video.parentNode.parentNode;
```

Comment:
for each parentNode, I would check that it exists before proceeding to get the next parentNode. II think we should also explicitly check that the second parentNode is a figure element, for example:

```
if (video.parentNode && video.parentNode.parentNode && video.parentNode.parentNode.classList.nodeName == 'figure')
```